File: 342-power-of-four/342-power-of-four.py
class Solution:
    def isPowerOfFour(self, n: int) -> bool:
        if n <= 0:
            return False
        elif n == 1:
            return True
        elif n < 1:
            return self.isPowerOfFour(1 / n)
        else:
            d = n / 4
            if int(d) != d:
                return False
            return self.isPowerOfFour(int(d))

File: 342-power-of-four/342-power-of-four.cpp
class Solution {
public:
    bool isPowerOfFour(int n) {
        if (n <= 0) {
            return false;
        }
        if (n == 1) {
            return true;
        }
        if (n < 1) {
            return this->isPowerOfFour(1 / n);
        }
        else {
            float d = (float)n / 4;
            if ((int)d != d) {
                return false;
            }
            return this->isPowerOfFour((int)d);
        }
    }
};